{
  try {
    return task.run(logger);
  }
 catch (  UnableToCompleteException e) {
  }
catch (  Throwable e) {
    logger.log(TreeLogger.ERROR,"Unexpected",e);
  }
  return false;
}
